fix(st-storage): fixing archived study configuration reading for short-term storages

MartinBelthle opened the pull request:
Cette Pull Request (PR) corrige un problème lié à la lecture des propriétés d'une étude lorsque celle-ci est archivée.

Plus précisément, l'erreur concerne la fonction de lecture de la configuration des stockages court terme, qui n'existe que pour les études dont la version est supérieure ou égale à 8.6. Lorsqu'on tentait de lire la configuration pour une étude archivée en version plus ancienne, on recevait le message d'erreur : "There is no item named 'input/st-storage/clusters/00_xp_batt_in/list.ini' in the archive".

Cette PR corrige la fonction `_parse_st_storage` pour qu'elle retourne une liste de stockages court terme vide lorsque l'étude est ancienne.

Une amélioration technique a également été apportée à la fonction d'extraction des fichiers compressés afin de mieux gérer les exceptions.

Les critères d'acceptation sont les suivants :

- Si l'étude n'est pas archivée, on doit voir la liste des stockages court terme (éventuellement vide si la version est inférieure à 8.6).
- Si l'étude est archivée et en version inférieure à 8.6, on doit obtenir une liste de stockage court terme vide.
- Si l'étude est archivée et en version supérieure ou égale à 8.6, on doit obtenir la liste des stockages court terme de l'étude (s'il y en a effectivement).

Actuellement, pour visualiser la liste des stockages court terme, il faudra utiliser la vue "Debug", car la vue "Stockages Court Terme" dédiée est encore en cours de développement.

MartinBelthle committed:
fix(st_storage): parse_st_storage works with zipped study with version previous 860

ghost commented:
### Bug Fixes

*  **api:** correct exception handling in `extract_file_to_tmp_dir` ([af38216](https://github.com/AntaresSimulatorTeam/AntaREST/commit/af38216e685b9f9d37e14cb66aa1300fe1cc592c))


### Code Refactoring

*  **api:** refactor data extraction logic in `_extract_data_from_file` function ([219990b](https://github.com/AntaresSimulatorTeam/AntaREST/commit/219990b14d3a4c9f64189fb7c9d245e9c0afba9c))


### Styles

* correct mypy issues ([8f5f97b](https://github.com/AntaresSimulatorTeam/AntaREST/commit/8f5f97b18298c27818b903c8f3475452dfdf5ad3))


### Contributors

<a href="https://github.com/MartinBelthle">MartinBelthle</a>, <a href="https://github.com/laurent-laporte-pro">laurent-laporte-pro</a>

<details>
<summary>Commit-Lint commands</summary>
<br />

You can trigger Commit-Lint actions by commenting on this PR:
- `@Commit-Lint merge patch` will merge dependabot PR on "patch" versions (X.X.Y - Y change)
- `@Commit-Lint merge minor` will merge dependabot PR on "minor" versions (X.Y.Y - Y change)
- `@Commit-Lint merge major` will merge dependabot PR on "major" versions (Y.Y.Y - Y change)
- `@Commit-Lint merge disable` will desactivate merge dependabot PR
- `@Commit-Lint review` will approve dependabot PR
- `@Commit-Lint stop review` will stop approve dependabot PR



</details>

laurent-laporte-pro reviewed (changes requested):
Attention, il faut soit attendre la correction de Takfarinas, soit supprimer son commit.

laurent-laporte-pro committed:
fix(api): correct exception handling in `extract_file_to_tmp_dir`

laurent-laporte-pro committed:
refactor(api): refactor data extraction logic in `_extract_data_from_file` function

- Add a concise docstring to the `_extract_data_from_file` function.
- Implement case-insensitive file suffix comparison to ensure robust file type detection.
- Consistently use UTF-8 encoding instead of platform-dependent encoding.
- Replace the custom `FileTypeNotSupportedException` exception with the standard `NotImplementedError`, simplifying the code.
- Refactor the implementation for shorter and more maintainable code, enhancing readability.

laurent-laporte-pro committed:
style: correct mypy issues
